src/activities/mirrorgame/mirrorgame.js

```diff
@@ -0,0 +1,526 @@
+
+                    if((x-w) === (z-y) && x!==w && y!==z )
+                     {
+
+                 if (component2.status === 1) {
+                 Light[count++] = component2.createObject(items.grid,{
+                                                      "x" : Math.max(y,z) * items.cellSize + items.cellSize/2+items.grid.spacing,
+                                                      "y" : Math.min(x,w) * items.cellSize + items.cellSize/2+items.grid.spacing,
+                                                       "height" : mod(y,z)*items.cellSize*1.414,
+                                                       "visible" : false,
+                                                        "uniquename" : "lightpath" +x+y+w+z
+                                                      }); }}
+
+
+
+                 }}}}
+
+}
+function createMirror(column, row) {
+
+
+    if (component3.status === 1) {
+         mirror[row][column] = component3.createObject(items.grid,
+                                                   {"x":column * items.cellSize+items.grid.spacing,
+                                                    "y":row * items.cellSize+2*items.grid.spacing,
+                                                    "width":items.cellSize,
+                                                    "height":items.cellSize
+                                                   })
+
+    } else {
+        console.log("error loading block component");
+        console.log(component3.errorString());
+        return false;
+    }
+    return true;
+}
+function createTorch(column, row) {
+
+
+    if (component4.status === 1) {
+         torch = component4.createObject(items.grid,
+                                                   {"x":column * items.cellSize+items.grid.spacing,
+                                                    "y":row * items.cellSize+items.grid.spacing,
+                                                    "width":items.cellSize,
+                                                    "height":items.cellSize
+                                                   })
+
+    } else {
+        console.log("error loading block component");
+        console.log(component4.errorString());
+        return false;
+    }
+    return true;
+}
+
+function createBulb(column, row) {
+
+    if (component5.status === 1) {
+         bulbon[num] = component5.createObject(items.grid,{"x":column * items.cellSize+items.grid.spacing,
+                                                    "y":row * items.cellSize+items.grid.spacing,
+                                                    "width":items.cellSize,
+                                                    "height":items.cellSize,
+                                                    "uniquename" : "bulbon" + row+column,
+                                                    "visible":false
+                                                   });
+
+    }
+        if (component6.status === 1) {
```

Comment:
you can probably have only one bulb array and lighten it when necessary (changing the image source)